app/Repositories/CircleRepository.php

```diff
@@ -19,8 +23,28 @@ public function getCircles($request) {
     }
 
     public function createCircle($request) {
-        return $this->model->create($request->only('name','token_name','team_sel_text','alloc_text','vouching',
-            'min_vouches','nomination_days_limit','vouching_text','team_selection','default_opt_in','only_giver_vouch'));
+        $data = $request->only('address', 'user_name', 'circle_name', 'protocol_id', 'protocol_name', 'uxresearch_json');
+        if(empty($data['protocol_id'])) {
+            $protocol = new Protocol(['name' => $data['protocol_name']]);
+            $protocol->save();
+            $protocol_id = $protocol->id;
+        } else {
+            $protocol_id = $data['protocol_id'];
+        }
+
+        $circle = $this->model->create(['name' => $data['circle_name'], 'protocol_id' => $protocol_id]);
+        $user = new User(['name' => $data['user_name'], 'circle_id' => $circle->id,
+            'role' => 1, 'address' => $data['address']]);
+        $user->save();
+        $profile = Profile::firstOrCreate([
+            'address' => $data['address']
+        ]);
+        $profile->load(['users.circle.protocol', 'users.teammates', 'users.histories.epoch']);
+
+        $research = new Uxresearch(['circle_id' => $circle->id, 'protocol_id' => $protocol_id,
```

Comment:
probably, if the user is already a circle admin, this data won't be present

Reply:
removed

Reply:
this is what i meant: https://github.com/coordinape/coordinape-backend/pull/47/commits/6f21ff6c553f19ffb3a89e98307017507c048939